src/containers/account/components/TransferList.js

```diff
@@ -1,88 +1,117 @@
-                >
-                    <MyGridTable
-                        data-testid={`transferListItem-${tx.transactionHash}`}
-                        key={`txTableDiv-${tx.blockNumber}-${tx.transactionIndex}-${tx.logIndex}-${tx.directionText}`}
-                    >
-                        <Row columns={1}>
-                            <Col>
-                                {tx.args.from.toLowerCase() === userAccountAddress.toLowerCase() ? (
-                                    <AccountAddress
-                                        address={tx.args.to}
-                                        showCopyIcon="true"
-                                        title="To:"
-                                        shortAddress={true}
-                                    />
-                                ) : (
-                                    <AccountAddress
-                                        address={tx.args.from}
-                                        showCopyIcon="true"
-                                        title="From:"
-                                        shortAddress={true}
-                                    />
-                                )}{" "}
-                                <MoreInfoTip header="Transaction details" icon="info" id={"transfer-" + index}>
-                                    blockNumber: {tx.blockNumber}
-                                    <br />blockHash: <small>{tx.blockHash}</small>
-                                    <br />Block timestamp: {tx.blockData.timestamp} {typeof tx.blockData.timestamp}
-                                    <br />transactionIndex: {tx.transactionIndex}
-                                    <br />
-                                    <HashURL hash={tx.transactionHash} type={"tx/"} />
-                                </MoreInfoTip>
-                            </Col>
-                        </Row>
-                        <Row columns={3}>
-                            <Col>Amount: {tx.signedAmount} A€</Col>
-                            <Col>Fee: {tx.senderFee} A€</Col>
-                            <Col>on {tx.blockTimeStampText}</Col>
-                        </Row>
+        const { header, noItemMessage, userAccount } = this.props;
+        const { isLoading, error } = this.props.transfers;
+        let { transfers } = this.props.transfers;
+
+        if (transfers) {
+            transfers = transfers.slice(0, this.state.limit);
 
-                        {tx.args.narrative && (
-                            <Row columns={1}>
-                                <Col>{tx.args.narrative}</Col>
-                            </Row>
-                        )}
-                    </MyGridTable>
-                </MyListGroup.Row>
-            ));
+            transfers.reduce((balance, tx, index, all) => {
+                const amount = index > 0 ? all[index - 1].signedAmount : 0;
+                return (tx.balance = Math.round(balance * 100 - amount * 100) / 100);
```

Comment:
All the time enough 2 digits?
(We can use  bignumber.js for the  float number calculation.)

Reply:
as there is only additions I think we don't need bignumbers here , rounding is enough (and A-EUR balances will stay in normal ranges unlike wei values)

Reply:
ok